crypto/cms/cms_sd.c

```diff
@@ -854,7 +854,7 @@ int CMS_SignerInfo_verify(CMS_SignerInfo *si)
 
     alen = ASN1_item_i2d((ASN1_VALUE *)si->signedAttrs, &abuf,
                          ASN1_ITEM_rptr(CMS_Attributes_Verify));
-    if (!abuf)
+    if (abuf == NULL || alen < 0)
```

Comment:
There seem to be other places that that either check the return value or the buffer, and only some that check both. I assume that if change this, you'll just get more reports that it's not checked consistently. Some of the other places check the return value for `<= 0`. Please at least document on how to check for an error of ASN1_item_i2d(), and then check for that.

Reply:
This has very little to do with all the other different cases of this call and is out of scope for this fix.
The real issue here is that the returned 'int' is then passed to a function as a 'size_t'.
Which is why the < 0 is needed. Other places that call this may not do the check if the code that uses the int does not try changing its type and just use it as a signed value.
As to documenting ASN1_item_i2d  - that sounds like a can of worms I don't want to eat here :)